### PR TITLE
Feature/military connected

### PR DESCRIPTION
**Pull request by stevemock**
data/backend changes to add the MCC

### CODE REVIEW

File: aggregate-service/src/main/resources/claim-report-template.sql.yml [hunk 1]
```diff
@@ -279,6 +293,11 @@ sqlbuilder:
                 where: >-
                   e.language_id in (:language_ids)
 
+            MilitaryConnectedCode_filter:
+              clauses:
+                where: >-
+                  e.military_connected_code in (:military_connected_code_ids)
```
Comment (malaffoon) [inline aggregate-service/src/main/resources/claim-report-template.sql.yml]: I'm pretty sure this needs to be `e.military_connected_id in (:military_connected_ids)`

File: aggregate-service/src/main/resources/custom-aggregate-report-template.sql.yml [hunk 2]
```diff
@@ -284,6 +298,11 @@ sqlbuilder:
                 where: >-
                   e.language_id in (:language_ids)
 
+            MilitaryConnectedCode_filter:
+              clauses:
+                where: >-
+                  e.military_connected_code in (:military_connected_code_ids)
```
Comment (malaffoon) [inline aggregate-service/src/main/resources/custom-aggregate-report-template.sql.yml]: ditto

File: aggregate-service/src/main/resources/longitudinal-report-template.sql.yml [hunk 3]
```diff
@@ -327,6 +341,11 @@ sqlbuilder:
                 where: >-
                   e.language_id in (:language_ids)
 
+            MilitaryConnectedCode_filter:
+              clauses:
+                where: >-
+                  e.military_connected_code in (:military_connected_code_ids)
```
Comment (malaffoon) [inline aggregate-service/src/main/resources/longitudinal-report-template.sql.yml]: ditto

File: aggregate-service/src/main/resources/target-report-template.sql.yml [hunk 4]
```diff
@@ -241,6 +255,11 @@ sqlbuilder:
                 where: >-
                   e.language_id in (:language_ids)
 
+            MilitaryConnectedCode_filter:
+              clauses:
+                where: >-
+                  e.military_connected_code in (:military_connected_code_ids)
```
Comment (malaffoon) [inline aggregate-service/src/main/resources/target-report-template.sql.yml]: ditto. Hmm, wonder if it is worth sharing some of these resources so we don't have to repeat them 4 times. I'll look into that at some point.

File: report-processor/src/test/java/org/opentestsystem/rdw/reporting/processor/repository/JdbcExamRepositoryIT.java [hunk 5]
```diff
@@ -332,6 +332,7 @@ private static String getField(String[] row, String name) {
             "LEPStatus",
```
Comment (malaffoon) [inline report-processor/src/test/java/org/opentestsystem/rdw/reporting/processor/repository/JdbcExamRepositoryIT.java]: Seems like there should be a change to a test proving the military connected code actually gets emitted.

File: reporting-service/src/main/java/org/opentestsystem/rdw/reporting/militaryconnected/DefaultMilitaryConnectedService.java [hunk 6]
```diff
@@ -0,0 +1,24 @@
...
+import javax.validation.constraints.NotNull;
+import java.util.List;
+
+@Service
+public class DefaultMilitaryConnectedService implements MilitaryConnectedService {
+
+    private final MilitaryConnectedRepository repository;
+
+    @Autowired
+    public DefaultMilitaryConnectedService(@NotNull final MilitaryConnectedRepository repository) {
```
Comment (malaffoon) [inline reporting-service/src/main/java/org/opentestsystem/rdw/reporting/militaryconnected/DefaultMilitaryConnectedService.java]: I guess you're copying existing style so i shouldn't comment but ... i don't know why we need `@NotNull` on an `@Autowired` method.

File: aggregate-service/src/test/java/org/opentestsystem/rdw/olap/service/impl/DefaultReportOptionServiceTest.java [hunk 7]
```diff
@@ -103,6 +108,7 @@ public void isShouldMapServiceResponsesToFieldsCorrectly() {
         final List<CodedEntity> genders = entities(id++);
         final List<SubjectAssessment> subjects = subjects(id++);
         final List<CodedEntity> languages = entities(id++);
+        final List<CodedEntity> militaryConnectedCode = entities(id++);
```
Comment (malaffoon) [inline aggregate-service/src/test/java/org/opentestsystem/rdw/olap/service/impl/DefaultReportOptionServiceTest.java]: nit: variable should probably be plural

File: reporting-service/src/main/java/org/opentestsystem/rdw/reporting/student/JdbcStudentRepository.java [hunk 8]
```diff
@@ -80,7 +80,8 @@
                             .individualEducationPlan(getNullable(row, row.getBoolean("iep")))
                             .limitedEnglishProficiency(getNullable(row, row.getBoolean("lep")))
                             .section504(getNullable(row, row.getBoolean("section504")))
-                            .migrantStatus(getNullable(row, row.getBoolean("migrant_status"))));
+                            .migrantStatus(getNullable(row, row.getBoolean("migrant_status")))
+                            .militaryConnectedCode(row.getString("military_connected_code")));
```
Comment (malaffoon) [inline reporting-service/src/main/java/org/opentestsystem/rdw/reporting/student/JdbcStudentRepository.java]: There should be a test change to go with this.

File: aggregate-service/src/main/resources/claim-report-template.sql.yml (hunk 1 above)
Comment (malaffoon) [reply aggregate-service/src/main/resources/claim-report-template.sql.yml]: I'll add a test for this.

File: aggregate-service/src/test/java/org/opentestsystem/rdw/olap/service/impl/DefaultReportOptionServiceTest.java (hunk 7 above)
Comment (stevemock) [reply aggregate-service/src/test/java/org/opentestsystem/rdw/olap/service/impl/DefaultReportOptionServiceTest.java]: Made plural

File: reporting-service/src/main/java/org/opentestsystem/rdw/reporting/student/JdbcStudentRepository.java (hunk 8 above)
Comment (stevemock) [reply reporting-service/src/main/java/org/opentestsystem/rdw/reporting/student/JdbcStudentRepository.java]: added it and languageCodes to a couple of the student constructs

**Pull request merged**